File: core/src/net/natruid/jungle/components/IndicatorOwnerComponent.java
package net.natruid.jungle.components;

import com.artemis.Component;
import com.artemis.annotations.PooledWeaver;
import com.badlogic.gdx.utils.ObjectMap;
import net.natruid.jungle.utils.PathNode;
import net.natruid.jungle.utils.types.IndicatorType;

@PooledWeaver
public class IndicatorOwnerComponent extends Component {
    public final ObjectMap<IndicatorType, PathNode[]> resultMap = new ObjectMap<>();
    public final ObjectMap<IndicatorType, int[]> indicatorMap = new ObjectMap<>();

    protected void reset() {
        resultMap.clear();
        indicatorMap.clear();
    }
}

File: core/src/net/natruid/jungle/components/IndicatorOwnerComponent.kt
package net.natruid.jungle.components

import com.artemis.PooledComponent
import net.natruid.jungle.utils.Area
import net.natruid.jungle.utils.IndicatorType

class IndicatorOwnerComponent : PooledComponent() {
    val resultMap = HashMap<IndicatorType, Area>()
    val indicatorMap = HashMap<IndicatorType, IntArray>()
    override fun reset() {
        resultMap.clear()
        indicatorMap.clear()
    }
}
